adapter-smoke-test/action.yml

```diff
@@ -31,3 +31,19 @@ runs:
     - name: Validate Podspec
       run: pod lib lint --verbose ${{ inputs.allow-warnings == 'true' && '--allow-warnings' || '' }}
       shell: bash
+
+    # Fix the CB SDK version in the podspec to the lowest one compatible with the adapter.
+    - name: Lower CB SDK Dependency Version
+      run: ruby "${{ github.action_path }}/../scripts/adapters/fix-podspec-cb-sdk-version-to-min.rb"
+      shell: bash
+
+    # Ensure adapter backward compatibility with the first minor version of the CB SDK major series.
+    - name: Ensure Backward Compatibility
+      run: pod lib lint --verbose ${{ inputs.allow-warnings == 'true' && '--allow-warnings' || '' }}
+      shell: bash
+
+    # Discard changes made to the podspec file
+    - name: Discard Podspec Changes
+      if: always()
+      run: ruby "${{ github.action_path }}/../scripts/adapters/discard-podspec-changes.rb"
+      shell: bash
```

Comment:
Would it be easier to combine all of these into one Ruby script, which is then called from a single step? It doesn't matter too much either way, but it just seems like it would be easier to do this all from one script.

Reply:
Yeah that's fair. I merged it into one script now.